rama-http/src/service/web/endpoint/response/octet_stream.rs

```diff
@@ -0,0 +1,174 @@
+        /// Set the filename for `Content-Disposition: attachment` header.
+        ///
+        /// This will trigger a file download in browsers with the specified filename.
+        pub fn file_name(mut self, filename: String) -> Self {
+            self.filename = Some(filename);
+            self
+        }
+    }
+
+    /// Convert into a partial content (HTTP 206) range response.
+    ///
+    /// This method consumes the `OctetStream` and converts it into a Response
+    /// with the specified byte range, including appropriate `Content-Range` and
+    /// `Accept-Ranges` headers. If a filename was set, the `Content-Disposition`
+    /// header will also be included.
+    ///
+    /// # Arguments
+    ///
+    /// * `range` - The byte range to serve (e.g., `0..5` for bytes 0-4)
+    /// * `complete_length` - The total length of the complete resource
+    ///
+    /// # Returns
+    ///
+    /// Returns `Ok(Response)` with status 206 (Partial Content) if the range is valid,
+    /// or `Err(Error)` if the range is invalid or out of bounds.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rama_http::service::web::response::OctetStream;
+    ///
+    /// # fn example() -> Result<(), rama_http_headers::Error> {
+    /// let data = b"Hello, World!".to_vec();
+    /// let total_len = data.len() as u64;
+    ///
+    /// // Serve bytes 0-4 (inclusive) out of 13 total bytes
+    /// let response = OctetStream::new(data)
+    ///     .try_into_range_response(0..5, total_len)?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn try_into_range_response(
+        self,
+        range: impl RangeBounds<u64>,
+        complete_length: u64,
+    ) -> Result<Response, Error>
+    where
+        T: Into<Body>,
+    {
+        let content_range =
+            ContentRange::bytes(range, complete_length).map_err(|_| Error::invalid())?;
+
+        let body = self.data.into();
+
+        let mut response = (
+            StatusCode::PARTIAL_CONTENT,
+            Headers((ContentType::octet_stream(), content_range)),
+            body,
+        )
+            .into_response();
+
+        // Add Content-Disposition if filename is provided
+        if let Some(filename) = self.filename {
+            response
+                .headers_mut()
+                .typed_insert(ContentDisposition::attachment(filename.as_str()));
+        }
+
+        Ok(response)
+    }
+}
+
+impl<T: fmt::Debug> fmt::Debug for OctetStream<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("OctetStream")
+            .field("data", &self.data)
+            .field("filename", &self.filename)
+            .finish()
+    }
+}
+
+impl<T: Clone> Clone for OctetStream<T> {
+    fn clone(&self) -> Self {
+        Self {
+            data: self.data.clone(),
+            filename: self.filename.clone(),
+        }
+    }
+}
+
+impl<T> IntoResponse for OctetStream<T>
+where
+    T: Into<Body>,
```

Comment:
These trait bounds will instead have to be something like this:

```
T: TryStream<Ok: Into<Bytes>, Error: Into<BoxError>> + Send + 'static,
```

Haven't tested it, so might miss some bounds or have a mistake.

Btw when taking a generic "Stream" we usually out of convention us the letter `S` instead of `T`.